### ISSUE TITLE
error occurred when run `cargo generate owner/repo`

### ISSUE THREAD

**Issue by pplmx** [cargo-generate/cargo-generate#752]
**Describe the bug**
CANNOT use the way of `owner/repo` directly.
While `cargo generate gh:<owner>/<repo>` or `cargo generate --git https://<repo_url>` are both fine

**make sure you**
- [x] I'm on the latest rust version `rustup update`
- [x] I'm on the latest cargo-generate version `cargo install cargo-generate`

**To Reproduce**
1. `cargo generate rust-cli/cli-template` or `cargo generate x-pt/template template/cxx`
2. 
```txt
⚠️   Favorite `x-pt/template` not found in config, using it as a git repository: x-pt/template
Error: Please check if the Git user / repository exists.

Caused by:
    unsupported URL protocol; class=Net (12)
```

```txt
⚠️   Favorite `rust-cli/cli-template` not found in config, using it as a git repository: rust-cli/cli-template
Error: Please check if the Git user / repository exists.

Caused by:
    unsupported URL protocol; class=Net (12)
```

**Expected behavior**
Like as 
- `cargo generate --git https://github.com/x-pt/template.git template/cxx`
- `cargo generate gh:x-pt/template template/cxx`

**Screenshots**
![image](https://user-images.githubusercontent.com/26994103/190178952-b07fbb5c-d557-4aff-b0ef-d691c737e7ce.png)
![image](https://user-images.githubusercontent.com/26994103/190179247-26274334-7fb9-4000-b68a-5ad3fb52d169.png)
![image](https://user-images.githubusercontent.com/26994103/190179335-384993cc-c6db-4eab-973e-881f0eb51f38.png)

**Desktop (please complete the following information):**
 - OS: [Windows]
 - Version: [win11]
 - Build tools: [msvc ]
 - Shell: [powerShell]
 - `rustc --version`: rustc 1.63.0 (4b91a6ea7 2022-08-08)
 - `cargo generate --version`: cargo-generate 0.16.0

**Additional context**

**Comment by taurr** [comment]
might very well be related to #674

**Comment by taurr** [comment]
might also be related to #664

**Comment by pplmx** [comment]
It seems `cargo generate <github-uname>/<repo-name>` does not be supported.
Tested on Ubuntu 22.04 in WSL2, which still failed.
```txt
❯ uname -a
Linux Aurora 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
❯ cat /etc/os-release
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
❯ cat /etc/debian_version
bookworm/sid
❯ cargo generate x-pt/template
⚠️   Favorite `x-pt/template` not found in config, using it as a git repository: x-pt/template
Error: Please check if the Git user / repository exists.

Caused by:
    unsupported URL protocol; class=Net (12)
```

**Comment by pplmx** [comment]
FYI.
But using this approach will cause many test cases to fail.
![image](https://user-images.githubusercontent.com/26994103/207293052-0dd80e67-05ca-4186-b9ba-0c2fa8a940cf.png)

**Comment by sassman** [comment]
@pplmx thanks for the suggestion, the issue is fixed in v0.17.4
